pandas/core/generic.py

```diff
@@ -2897,6 +2902,10 @@ def to_latex(
         if multirow is None:
             multirow = config.get_option("display.latex.multirow")
 
+        # formatters is a list
+        if formatters_col:
+            formatters = [lambda x: style % x for style in formatters_col]
```

Comment:
Is it possible to use an f-string here instead?

Reply:
Also could you add a check to validate that `formatters_col` is a list (or an iterable)?

Reply:
I've added a check that silently fails if it's not a list, not sure if this is the best way to do this (feels un-pythonic but I digress)

Reply:
I attempted an f-string approach as such: 
```formatters = [lambda x: f"{style}{x}" for style in formatters_col]``` 
But I was failing the test suite (broke functionality), maybe there's a better approach I'm not seeing?